lib/tori/backend/s3.rb

```diff
@@ -91,6 +91,17 @@ def read(filename)
         body(filename).read
       end
 
+      def public_url(filename)
+        "#{client.config.endpoint}/#{@bucket}/#{filename}"
+      end
+
+      def url_for(filename, method)
+        signer = Aws::S3::Presigner.new(client: client)
+        signer.presigned_url(method, bucket: @bucket, key: filename)
+      end
+
+      private
+
```

Comment:
You should not think too much.
Common methods are exists.
Special methods by backend are also exists.

If you want to use both backend, I propose `case` expression.

```ruby
def url(tori_file)
  case Tori.config.backend
  when Tori::Backend::FileSystem
    # use static file for development
    "http://localhost:3000/tori/#{tori_file.name}"
  when Tori::Backend::S3
    # use s3 file for production
    tori_file.public_url
  end
end
```